galaxy_importer/loaders/collection.py

```diff
@@ -99,6 +105,58 @@ def load(self):
             requires_ansible=self.requires_ansible,
         )
 
+    def _lint_collection(self):
+        """Log ansible-lint output.
+
+        ansible-lint stdout are linter violations, they are logged as warnings and errors,
+        depending on the rule level.
+
+        ansible-lint stderr includes info about vars, file discovery,
+        summary of linter violations, config suggestions, and raised errors.
+        Only raised errors are logged, they are logged as errors.
+        """
+
+        self.log.info("Linting collection via ansible-lint...")
+
+        if not shutil.which("ansible-lint"):
+            self.log.warning("ansible-lint not found, skipping lint of collection")
+            return
+
+        cmd = [
+            "/usr/bin/env",
+            f"ANSIBLE_LOCAL_TEMP={self.cfg.ansible_local_tmp}",
+            "ansible-lint",
+            "--profile",
+            "production",
+            "--exclude",
+            "tests/integration/",
+            "--exclude",
+            "tests/unit/",
+            "--parseable",
+            "--nocolor",
+        ]
+        self.log.debug("CMD: " + " ".join(cmd))
+        proc = Popen(
+            cmd,
+            cwd=self.path,
+            encoding="utf-8",
+            stdout=PIPE,
+            stderr=PIPE,
+        )
+        proc.wait()
```

Comment:
We'll need to replace `proc.wait()` with `proc.communicate()` to prevent deadlocks. Right now I believe deadlocks are very rare, but when calling lint on the entire collection instead of one role at a time, it will likely increase. You can replicate the change made in #213

Reply:
Change replicated.